qt/scientific_interfaces/Direct/ALFView_model.cpp

```diff
@@ -0,0 +1,124 @@
+#include "ALFView_model.h"
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/NumericAxis.h"
+
+#include "MantidGeometry/Instrument.h"
+
+#include "MantidKernel/Unit.h"
+
+#include <utility>
+
+namespace {
+const std::string tmpName = "ALF_tmp";
+const std::string instrumentName = "ALF";
+const std::string wsName = "ALFData";
+} // namespace
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+void ALFView_model::loadEmptyInstrument() {
+  Mantid::API::IAlgorithm_sptr alg =
+      Mantid::API::AlgorithmManager::Instance().create("LoadEmptyInstrument");
+  alg->initialize();
+  alg->setProperty("OutputWorkspace", wsName);
+  alg->setProperty("InstrumentName", instrumentName);
+  alg->execute();
+}
+/*
+ * Loads data for use in ALFView
+ * Loads data, normalise to current and then converts to d spacing
+ * @param name:: string name for ALF data
+ * @return int:: the run number
+ */
+int ALFView_model::loadData(const std::string &name) {
+  Mantid::API::IAlgorithm_sptr alg =
+      Mantid::API::AlgorithmManager::Instance().create("Load");
+  alg->initialize();
+  alg->setProperty("Filename", name);
+  alg->setProperty("OutputWorkspace", tmpName); // write to tmp ws
+  alg->execute();
+  Mantid::API::MatrixWorkspace_sptr ws =
```

Comment:
Similar here. `auto ws = ` and `using `Mantid::API::AnalysisDataService` above would improve readability.